portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,20 +27,31 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
 
 /** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxComments;
+    try {
+        maxComments = Integer.parseInt(request.getParameter("max-comments"));
+    }
+    catch (Exception e) {
+        // Default to displaying 3 comments in case of error.
+        maxComments = 3;
```

Comment:
Another way to handle a bad input, is to not execute the rest of the servlet and return an HTTP error. Not required for this starter project but something useful that you will probably use a lot in the capstone.

https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/http/HttpServletResponse.html#sendError(int,%20java.lang.String)